File: pset6/sentimental-mario-more/mario.py
# TODO
from cs50 import get_int

while True:
    h = get_int("Height: ")
    if h < 1 or h > 8:
        print("Height must be between 1 and 8")
    else:
        break

for i in range(h):
    print(" " * (h - i - 1) + "#" * (i + 1) + "  " + "#" * (i + 1))

File: pset1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_fn(int cnt, char type);

int main(void)
{
    // get user input
    int x;
    do
    {
        x = get_int("Height: ");
    } while (x < 1 || x > 8);

    // for each row
    for (int i = 0; i < x; i++)
    {
        int space_cnt = x - i - 1;
        int hash_cnt = i + 1;

        // for each col
        print_fn(space_cnt, ' ');
        print_fn(hash_cnt, '#');
        print_fn(2, ' ');
        print_fn(hash_cnt, '#');
        printf("\n");
    }
}

void print_fn(int cnt, char type)
{
    for (int i = 0; i < cnt; i++)
    {
        printf("%c", type);
    }
}
